apps/print/backend/src/util/ballot_styles.ts

```diff
@@ -0,0 +1,77 @@
+import type {
+  Election,
+  BallotStyleId,
+  Id,
+  PrecinctId,
+  LanguageCode,
+} from '@votingworks/types';
+
+import { assert, find, throwIllegalValue } from '@votingworks/basics';
+import { getAllPrecinctsAndSplits } from '@votingworks/types';
+import { getBallotStyleGroupsForPrecinctOrSplit } from '@votingworks/utils';
+
+interface FindBallotStyleArgs {
+  precinctId: PrecinctId;
+  splitId?: Id;
+  languageCode: LanguageCode;
+  partyId?: Id;
+}
+
+export function findBallotStyleId(
```

Comment:
The logic here roughly follows VxMark's ballot style selection [logic](https://github.com/votingworks/vxsuite/blob/jonah/refine-mark-finding/libs/mark-flow-ui/src/components/poll_worker/ballot_style_select.tsx#L51), although we must also consider language and party